exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/PostgreSQL.kt

```diff
@@ -16,11 +16,22 @@ internal object PostgreSQLDataTypeProvider : DataTypeProvider() {
         exposedLogger.warn("The length of the binary column is not required.")
         return binaryType()
     }
-    override fun blobType(): String = "bytea"
+    override fun blobType(): String = if (currentDialect is PostgreSQLDialect) "oid" else "bytea"
     override fun uuidToDB(value: UUID): Any = value
     override fun dateTimeType(): String = "TIMESTAMP"
     override fun jsonBType(): String = "JSONB"
     override fun hexToDb(hexString: String): String = """E'\\x$hexString'"""
+
+    override fun processForDefaultValue(e: Expression<*>): String {
+        val value = super.processForDefaultValue(e)
+        if (e is LiteralOp<*> &&
+            e.columnType is BlobColumnType &&
+            currentDialect is PostgreSQLDialect) {
+            return "lo_from_bytea(0, $value :: bytea)"
+        }
```

Comment:
Please add the new column property and check for the absence of H2 modes instead:
```kt
e is LiteralOp<*> && e.columnType is BlobColumnType && e.columnType.useObjectIdentifier && (currentDialect as? H2Dialect) == null -> {
    "lo_from_bytea(0, ${super.processForDefaultValue(e)} :: bytea)"
}
```